src/server/routes/shorten/shorten.ts

```diff
@@ -18,16 +18,23 @@ import { Request, Response, Router } from "express";
 import * as request from "request-promise-native";
 import { SettingsGetter } from "../../utils/settings-manager/settings-manager";
 
+const getIpAddressFromRequest = (req: Request) => {
+  const ip = req.headers["x-forwarded-for"] || req.connection.remoteAddress;
+
+  return typeof ip === "string" ? ip.split(",").pop().trim() : ip[0];
```

Comment:
Are we sure that it’s non empty array if it’s not a string? More defensive checks would be great.

Reply:
The check for `trustProxy` made it even easier, because according to [the docs](https://expressjs.com/en/api.html#req.ip) `req.ip` does the work for us when the proxy is set, and otherwise we can rely on `req.connection.remoteAddress`.